piptools/resolver.py

```diff
@@ -170,7 +170,8 @@ def resolve(self, max_rounds=10):
 
             log.debug("")
             log.debug(magenta("{:^60}".format("ROUND {}".format(current_round))))
-            has_changed, best_matches = self._resolve_one_round()
+            with self.repository:
```

Comment:
```suggestion
            # If a package version (foo==2.0) was built in a previous round,
            # and in this round a different version of foo needs to be built
            # (i.e. foo==1.0), the directory will exist already, which will
            # cause a pip build failure.  The trick is to start with a new
            # build cache dir for every round, so this can never happen.
            with self.repository.freshen_build_caches():
```

I would keep the comment — it explains the intention. And I think it makes sense to call an explicit context manager in favor of readability (minor).

Reply:
Thanks for the review. I made this change in the latest revision.